basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -139,153 +73,115 @@ public void removeSecurityRealm(String realmName) throws CommandFailedException,
-        executeAddAction(address, Collections.emptyMap());
-    }
-
-    private void executeAddAction(ResourceAddress address, boolean reload) throws InterruptedException, TimeoutException, IOException {
-        executeAddAction(address, Collections.emptyMap(), reload);
-    }
-
-    public String createWorker() throws InterruptedException, TimeoutException, IOException {
-        String name = "UndertowWorker_" + RandomStringUtils.randomAlphanumeric(6);
-        ResourceAddress address = ioWorkerAddressTemplate.resolve(context, name);
-        executeAddAction(address);
-        return name;
+        return bufferPool;
     }
 
-    public void removeWorker(String name) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = ioWorkerAddressTemplate.resolve(context, name);
-        executeRemoveAction(address);
+    public void removeBufferPool(String name) throws IOException {
+        Address address = Address.subsystem("io").and("buffer-pool", name);
+        operations.remove(address);
     }
 
-    public String createBufferPool() throws InterruptedException, TimeoutException, IOException {
-        String name = "UndertowBufferPool_" + RandomStringUtils.randomAlphanumeric(6);
-        ResourceAddress address = ioBufferPoolAddressTemplate.resolve(context, name);
-        executeAddAction(address);
+    public String createSocketBinding(String name) throws IOException, CommandFailedException {
+        int port = AvailablePortFinder.getNextAvailable();
+        log.info("Obtained port for socket binding '" + name + "' is " + port);
+        client.apply(new AddSocketBinding.Builder(name)
+                .port(port)
+                .build());
         return name;
     }
 
-    public void removeBufferPool(String name) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = ioBufferPoolAddressTemplate.resolve(context, name);
-        executeRemoveAction(address);
-    }
-
     public String createSocketBinding() throws CommandFailedException, IOException {
         String name = "UndertowSocketBinding_" + RandomStringUtils.randomAlphanumeric(6);
-        try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
-            int port = AvailablePortFinder.getNextAvailable();
-            log.info("Obtained port for socket binding '" + name + "' is " + port);
-            client.apply(new AddSocketBinding.Builder(name)
-                    .port(port)
-                    .build());
+        return createSocketBinding(name);
+    }
+
+    /**
+     * If keepReference is set to true, name of created socket binding name will be kept internally in UndertowOperations
+     * @param keepReference whether the name of newly created socket binding will be saved
+     */
+    public String createSocketBinding(boolean keepReference) throws IOException, CommandFailedException {
```

Comment:
Is `keepReference` argument necessary? May be you can rename this method to something like `createSocketBindingWithReference` or `createAutoCleanableSocketBinding` since anyway `createSocketBinding(false)` is just an alias for `createSocketBinding()`